NodeJS/week-2/Homework/app.js

```diff
@@ -0,0 +1,81 @@
+import express from "express";
+import fs from "fs";
+
+const app = express();
+const port = process.env.PORT || 3000;
+
+// Support parsing JSON requests
+app.use(express.json());
+
+// Load documents from JSON file
+let documents = JSON.parse(fs.readFileSync("documents.json", "utf-8"));
+
+// GET /
+app.get("/", (req, res) => {
+  res.send("This is a search engine");
+});
+
+// GET /search - Search for documents based on query parameter q
+app.get("/search", (req, res) => {
+  const query = req.query.q;
+
+  if (!query) {
+    return res.json(documents); // Return all documents if no query
+  }
+
+  // Filter documents based on any field that matches the query
+  const results = documents.filter((doc) =>
+    Object.values(doc).some(
+      (value) => typeof value === "string" && value.includes(query)
+    )
+  );
```

Comment:
Instead of using Object.values(document) it could be a good idea to explicitly choose the fields/attributes which should be searched. Then you are also making sure that you don't have to check for a string value